eng/pipelines/templates/jobs/ci.yml

```diff
@@ -47,301 +47,82 @@ parameters:
-              /p:ArtifactsPackagesDir=$(Build.ArtifactStagingDirectory) `
-              /p:ServiceDirectory=* `
-              /p:ProjectListOverrideFile=$(ProjectListOverrideFile) `
-              /p:IncludePerf=false `
-              /p:IncludeStress=false `
-              /p:IncludeIntegrationTests=false `
-              /p:EnableOverrideExclusions=true `
-              $(DiagnosticArguments)
-          displayName: "Build and Package for PR"
-      - ${{ else }}:
-        - pwsh: |
-            $variableSet = $env:ProjectListOverrideFile -ne "`$`(ProjectListOverrideFile`)"
-            if ($variableSet -and (Test-Path $env:ProjectListOverrideFile)) {
-              Write-Host "Clearing $env:ProjectListOverrideFile"
-              rm $env:ProjectListOverrideFile
-            }
-            Write-Host '##vso[task.setvariable variable=ProjectListOverrideFile;]'
-          env:
-            ProjectListOverrideFile: $(ProjectListOverrideFile)
-          displayName: Cleanup Props File
-
-        - script: >-
-            dotnet pack eng/service.proj -warnaserror
-            /p:ValidateRunApiCompat=true
-            /p:SDKType=${{ parameters.SDKType }}
-            /p:ServiceDirectory=${{ parameters.ServiceDirectory }}
-            /p:IncludeTests=false
-            /p:PublicSign=false $(VersioningProperties)
-            /p:Configuration=$(BuildConfiguration)
-            /p:CommitSHA=$(Build.SourceVersion)
-            /p:ArtifactsPackagesDir=$(Build.ArtifactStagingDirectory)
-            /p:ProjectListOverrideFile=$(ProjectListOverrideFile)
-            $(DiagnosticArguments)
-          displayName: "Build and Package Service Directory"
-
-      # save package properties again, this time honoring the dev version number
-      - ${{ if ne(parameters.ServiceDirectory, 'auto') }}:
-        - template: /eng/common/pipelines/templates/steps/save-package-properties.yml
-          parameters:
-            ServiceDirectory: ${{ parameters.ServiceDirectory }}
-
-      - task: Powershell@2
-        inputs:
-          filePath: $(Build.SourcesDirectory)/eng/scripts/Save-Package-Namespaces-Property.ps1
-          arguments: >
-            -RepoRoot $(Build.SourcesDirectory)
-            -PackageInfoDirectory $(Build.ArtifactStagingDirectory)/PackageInfo
-            -BuildConfiguration $(BuildConfiguration)
-          pwsh: true
-          workingDirectory: $(Pipeline.Workspace)
-        displayName: Update package properties with namespaces
-        condition: and(succeeded(), ne(variables['ProjectNames'], ''))
-
-      - template: ../steps/archetype-sdk-docs.yml
-        parameters:
-          ServiceDirectory: ${{parameters.ServiceDirectory}}
-          Artifacts: ${{parameters.Artifacts}}
-          DocGenerationDir: "$(Build.SourcesDirectory)/doc/ApiDocGeneration"
-          LibType: 'client'
-          PackageInfoFolder: '$(Build.ArtifactStagingDirectory)/PackageInfo'
+      variables:
+        Codeql.Enabled: true
```

Comment:
It is kind of annoying these are copied I wonder if it is worth pulling out into a variable/build.variables.yml file?

Reply:
Yeah fair point and I legit thought about it already. You bringing it up just means I'll execute 👍

Reply:
I have copied the variables that are set statically, the ones set by parameter will have to remain.